### ISSUE TITLE
False positive in testset with anonymous function

### ISSUE THREAD

**Issue by schillic** [julia-vscode/StaticLint.jl#362]
The following reports `Cannot define function ; it already has a value.` in the line marked with a comment. But running the code works fine because, as the *Note* box [here](https://docs.julialang.org/en/v1/stdlib/Test/#Working-with-Test-Sets) says, `@testset` creates a new scope. There is no warning when writing `f(x) = 1` instead of the anonymous function in the first block.

```julia
using Test

@testset "1" begin
    f = x -> 1
    f(1)
end

@testset "2" begin
    f(x) = 2  # <- warning here
    f(2)
end
```

**Comment by pfitzseb** [comment]
This is fixed now.
